src/service/owm/weather.cpp

```diff
@@ -149,9 +149,16 @@ void weather_start() {
 
     firstRunSemaphore = xSemaphoreCreateBinary();
 
-    apiKey = prefs_get_string(PREFS_KEY_WEATHER_APIKEY, String(WEATHER_API_KEY));
-    latitude = prefs_get_string(PREFS_KEY_WEATHER_LAT, String(WEATHER_LAT));
-    longitude = prefs_get_string(PREFS_KEY_WEATHER_LON, String(WEATHER_LON));
+    apiKey = prefs_get_string(PREFS_KEY_WEATHER_APIKEY);
+    latitude = prefs_get_string(PREFS_KEY_WEATHER_LAT);
+    longitude = prefs_get_string(PREFS_KEY_WEATHER_LON);
+
+    if (!apiKey.length()){
+        ESP_LOGW(LOG_TAG, "No weather API key, disabling weather updates!");
+        return;
+    } else {
+        ESP_LOGI(LOG_TAG, "Weather API key: %s", apiKey.c_str());
+    }
```

Comment:
If possible, sensitive data should not be logged. (Even though it's currently visible via WebUI, that might [and should] eventually change)